### DISCUSSION TITLE
How to list fonts used in a PDF file?

### DISCUSSION THREAD

**Discussion by gmile** [pdfcpu/pdfcpu#986] [Q&A] [answered]
Is there a way to list fonts installed in a PDF? I can't seem to find it. I tried `pdfcpu fonts`, but appears it's used for a different purpose.

Today, I can use a tool called `pdffonts` to list fonts. Here's the kind of output I get:

```
$ pdffonts ~/Downloads/broken.pdf
name                                 type              encoding         emb sub uni object ID
------------------------------------ ----------------- ---------------- --- --- --- ---------
BCDEEE+Calibri                       TrueType          WinAnsi          yes yes no       9  0
Arial-BoldMT                         TrueType          WinAnsi          no  no  no      13  0
ArialMT                              TrueType          WinAnsi          no  no  no      15  0
BCDFEE+Wingdings-Regular             CID TrueType      Identity-H       yes yes yes     23  0
ArialMT                              CID TrueType      Identity-H       yes no  yes     28  0
Arial-BoldMT                         CID TrueType      Identity-H       yes no  yes     71  0
BCDGEE+Century                       CID TrueType      Identity-H       yes yes yes     76  0
```

I'd like to know if there's currently way to get the same list using `pdfcpu`, preferably with an option to output a JSON (since it's not possible to output `json` via `pdftools`, see [this](https://gitlab.freedesktop.org/poppler/poppler/-/issues/1341#note_2371593)).

For context, I'm looking at `pdfcpu` as a single replacement tool for a set of tools I'm using currently. Ability to analyze which fonts the PDF is using and whether the fonts are embedded or not can help me return better errors to users who are uploading PDFs to my application.

**Comment by hhrutter** [comment, marked as answer]
`pdfcpu validate -v in.pdf` will produce output that shows all fonts used.

**Comment by gmile** [comment]
@hhrutter great, thanks! I didn't know `validate` returns this info! I just tried it and can see a section called "Fonts:" which looks like what I am after:

```
...
Fonts:
obj     prefix     Fontname                       Subtype    Encoding             Embedded ResourceIds
#13                Arial-BoldMT                   TrueType   WinAnsiEncoding      true     F2
#71                Arial-BoldMT                   Type0      Identity-H           true     F6
#15                ArialMT                        TrueType   WinAnsiEncoding      true     F3
#28                ArialMT                        Type0      Identity-H           true     F5
#9      BCDEEE     Calibri                        TrueType   WinAnsiEncoding      true     F1
#76     BCDGEE     Century                        Type0      Identity-H           true     F7
#23     BCDFEE     Wingdings-Regular              Type0      Identity-H           true     F4
...
```

Is it possible to make it return the info in JSON format? The reason JSON format is useful is to deal with situations when a font name contains a space, for example. In such a case, parsing a table output becomes much harder.

**Comment by hhrutter** [comment]
Sure, I just need to think about in which way eg. there already is `pdfcpu info -json in.pdf` we could expand on that or maybe a separate command.

Can you do me a favor and open a feature request issue?
Thx

**Reply by gmile** [reply]
Sure, opened this one: https://github.com/pdfcpu/pdfcpu/issues/987. Also, marked your reply above as an answer. Thank you!
